container/container_test.go

```diff
@@ -0,0 +1,417 @@
+			Convey("when the list is non-empty", func() {
+				cntList, err := newOperator.GetNewContainers(ctx)
+				So(err, ShouldBeNil)
+				So(len(cntList), ShouldEqual, 2)
+
+				err = newOperator.RememberCurrentContainers(ctx)
+				So(err, ShouldBeNil)
+
+				cntList, err = newOperator.GetNewContainers(ctx)
+				So(err, ShouldBeNil)
+				So(len(cntList), ShouldEqual, 0)
+			})
+
+			Convey("not when the list is empty", func() {
+				cntList, errc := empNewOperator.GetNewContainers(ctx)
+				So(errc, ShouldNotBeNil)
+				So(len(cntList), ShouldEqual, 0)
+			})
+		})
+
+		Convey("it can check if the container name is valid", func() {
+			clist, err := newOperator.GetCurrentContainers(ctx)
+			So(err, ShouldBeNil)
+			So(len(clist), ShouldEqual, 3)
+
+			container1 := clist[0]
+			Convey("for a correct name of the container", func() {
+				So(container1.HasName("test_container1_2"), ShouldBeTrue)
+			})
+
+			Convey("but not for a wrong name of the container", func() {
+				So(container1.HasName("wrong_name"), ShouldBeFalse)
+			})
+		})
+
+		Convey("it can get the new container given a name", func() {
+			Convey("when the list of containers is non-empty", func() {
+				Convey("and the new container name is given", func() {
+					cntr, err := newOperator.GetNewContainerByName(ctx, "test_container2")
+					So(cntr, ShouldNotBeNil)
+					So(err, ShouldBeNil)
+				})
+
+				Convey("but not when an old container name is given", func() {
+					cntr, err := newOperator.GetNewContainerByName(ctx, "test_container3")
+					So(cntr, ShouldBeNil)
+					So(err, ShouldBeNil)
+				})
+
+				Convey("but not when an non-existing container name is given", func() {
+					cntr, err := newOperator.GetNewContainerByName(ctx, "wrong_name")
+					So(cntr, ShouldBeNil)
+					So(err, ShouldBeNil)
+				})
+			})
+
+			Convey("not when the list of containers is empty", func() {
+				cntr, err := empNewOperator.GetNewContainerByName(ctx, "wrong_name")
+				So(cntr, ShouldBeNil)
+				So(err, ShouldNotBeNil)
+			})
+		})
+
+		Convey("it can get a container given the id", func() {
+			Convey("for a client with a non-empty list of containers", func() {
+				Convey("when the container id is correct", func() {
+					cntr, err := newOperator.GetContainerByID(ctx, "container_id1")
+					So(cntr, ShouldNotBeNil)
+					So(err, ShouldBeNil)
+				})
+
+				Convey("when the container id is wrong", func() {
+					cntr, err := newOperator.GetContainerByID(ctx, "wrong_id")
+					So(cntr, ShouldBeNil)
+					So(err, ShouldBeNil)
+				})
+			})
+
+			Convey("but not for a client with an empty list of containers", func() {
+				cntr, err := empNewOperator.GetContainerByID(ctx, "wrong_id")
+				So(cntr, ShouldBeNil)
+				So(err, ShouldNotBeNil)
+			})
+		})
+
+		Convey("and given a file path/glob path, return the container id", func() {
+			// Create some files containing container id
+			containerTempDir, err := ioutil.TempDir("", "container_temp_")
```

Comment:
After checking for err, you should `defer os.RemoveAll(containerTempDir)`

Reply:
Added